packages/menu/src/Menu/Menu.tsx

```diff
@@ -185,14 +188,19 @@ export const Menu = React.forwardRef<HTMLDivElement, MenuProps>(function Menu(
       >
         <Popover
           key="popover"
+          ref={popoverRef}
           active={open}
           align={align}
           justify={justify}
           refEl={refEl}
           adjustOnMutation={adjustOnMutation}
+          // Need to stop propagation, otherwise Menu will closed automatically when clicked
+          onClick={e => e.stopPropagation()}
+          onKeyDown={handleKeyDown}
           onEntered={handlePopoverOpen}
           data-testid={LGIDs.root}
           data-lgid={LGIDs.root}
+          {...rest}
```

Comment:
Could this rest override the `onClick` above?

Reply:
Oo good catch!